cmd/checkConnectivity.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE file in the project root for license information.
+
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/Azure/kubectl-az/cmd/utils"
+	"github.com/spf13/cobra"
+)
+
+var (
+	resourceGroup string
+	cluster       string
+)
+
+var checkConnectivityCmd = &cobra.Command{
+	Use:          "check-connectivity",
+	Short:        "Check connectivity between the nodes and the Kubernetes API Server",
+	RunE:         checkConnectivityCmdRun,
+	SilenceUsage: true,
+}
+
+func init() {
+	checkConnectivityCmd.Flags().StringVarP(
+		&resourceGroup,
+		"resource-group", "",
```

Comment:
Instead of resource-group and cluster can the check just take the kubernetes node name? Also, if no node parameter is provided, it could mean to enumerate through all node objects in the cluster checking the connectivity of each.

Reply:
Please check [this discussion](https://github.com/Azure/kubectl-az/pull/1#discussion_r796798441).